docs/zkapps/tutorials/09-recursion.mdx

```diff
@@ -56,86 +56,310 @@ You build recursive zkApps with [ZkProgram](/zkapps/o1js-reference/modules/Exper
 
-To create a ZkProgram, start with the `init()` method. 
+:::info
 
-- For each method, declare the inputs it will receive. 
-- The first argument of a ZkProgram method is always the state of the ZkProgram, named `publicInput` since it is public.
+To prevent copying line numbers and command prompts as shown in the examples, use the copy code to clipboard button that appears at the top right of the snippet box when you hover over it.
 
-```typescript
-const Add = ZkProgram({
-  name: 'add-example',
-  publicInput: Field,
+:::
 
-  methods: {
-    init: {
-      privateInputs: [],
+## Prerequisites
+
+Ensure your environment meets the [Prerequisites](/zkapps/tutorials#prerequisites) for zkApp Developer Tutorials.
+
+In particular, make sure you have the zkApp CLI installed:
+
+```sh
+$ npm install -g zkapp-cli
+```
+
+## Create a new project
+
+Now that you have the tooling installed, you can start building your application.
+
+1. Create or change to a directory where you have write privileges.
+1. Now, create a project using the `zk project` command:
+
+```sh
+$ zk project 09-recursion
+```
+
+The `zk project` command has the ability to scaffold the UI for your project. For this tutorial, select `none`:
+
+```
+? Create an accompanying UI project too? …
+  next
+  svelte
+  nuxt
+  empty
+❯ none
+```
+
+The expected output is:
+
+```sh
+✔ Create an accompanying UI project too? · none
+✔ UI: Set up project
+✔ Initialize Git repo
+✔ Set up project
+✔ NPM install
+✔ NPM build contract
+✔ Set project name
+✔ Git init commit
+
+Success!
 
-      method(state: Field) {
-        state.assertEquals(Field(0));
-      },
-    },
+Next steps:
+  cd 09-recursion
+  git remote add origin <your-repo-url>
+  git push -u origin main
+```
+
+The `zk project` command creates the `09-recursion` directory that contains the scaffolding for your project, including tools such as the Prettier code formatting tool, the ESLint static code analysis tool, and the Jest JavaScript testing framework.
+
+1. Change into the `09-recursion` directory and list the contents:
+
+```sh
+$ cd 09-recursion
+$ ls
+```
+
+The output shows these results:
+
+```sh
+LICENSE
+README.md
+babel.config.cjs
+build
+config.json
+jest-resolver.cjs
+jest.config.js
+keys
+node_modules
+package-lock.json
+package.json
+src
+tsconfig.json
+```
+
+For this tutorial, you run commands from the root of the `09-recursion` directory as you work in the `src` directory on files that contain the TypeScript code for the smart contract. Each time you make updates, then build or deploy, the TypeScript code is compiled into JavaScript in the `build` directory.
+
+### Prepare the project
+
+Start by deleting the default files that come with the new project.
+
+1. To delete the old files:
+
+```sh
+$ rm src/Add.ts
+$ rm src/Add.test.ts
+$ rm src/interact.ts
+```
+
+1. Now, create the new files for your project:
+
+```sh
+$ zk file src/Add
```

Comment:
did you want to use Square.ts? https://github.com/o1-labs/docs2/blob/main/examples/zkapps/09-recursion/src/Square.ts

Reply:
Hi @barriebyron ,

 I went through all the files and I am not sure why `Square.ts` is used here. Code inside this file does not aligns to what  tutorial is actually for. 
According to me we can remove it, if possible can we ask someone from team to review it ? 

Same is for `zkProgram_wrapper.ts`.

Can you please help to get it reviewed so I can remove it in coming commits ?

Reply:
About `Add.ts` I will replace it with the original one from previous tutorial i.e `main.ts`.